java/code/src/com/suse/manager/webui/utils/SparkApplicationHelper.java

```diff
@@ -550,33 +556,280 @@ public static JadeTemplateEngine setup() {
+     * @param response the http response
+     * @param result the object to serialize to JSON
+     * @param httpStatusCode http status code
+     * @param type type token needed for gson
+     * @param <T> type of the result
+     * @return a JSON string
+     */
+    public static <T> String json(Response response, int httpStatusCode, T result, TypeToken<T> type) {
+        response.type("application/json");
+        response.status(httpStatusCode);
+        return GSON.toJson(result, type.getType());
+    }
+
+    /**
+     * Serialize the result and set the response content type to JSON.
+     * @param response the http response
+     * @param result the object to serialize to JSON
+     * @param type type token needed for gson
+     * @param <T> type of the result
+     * @return a JSON string
+     */
+    public static <T> String result(Response response, ResultJson<T> result, TypeToken<T> type) {
+        response.type("application/json");
+        ParameterizedType parameterizedType = newParameterizedTypeWithOwner(null, ResultJson.class, type.getType());
+        return GSON.toJson(result, parameterizedType);
+    }
+
+    /**
+     * Serialize the result and set the response content type to JSON.
+     * @param gson {@link Gson} object to use for serialization
+     * @param response the http response
+     * @param result the long to serialize to JSON
+     * @return a JSON string
+     */
+    public static String json(Gson gson, Response response, long result) {
+        response.type("application/json");
+        return gson.toJson(result);
+    }
+
+    /**
+     * Serialize the result and set the response content type to JSON.
+     * @param gson {@link Gson} object to use for serialization
+     * @param response the http response
+     * @param result the object to serialize to JSON
+     * @param type type token needed for gson
+     * @param <T> type of the result
+     * @return a JSON string
+     */
+    public static <T> String json(Gson gson, Response response, T result, TypeToken<T> type) {
+        response.type("application/json");
+        return gson.toJson(result, type.getType());
+    }
+
     /**
      * Serialize the result and set the response content type to JSON
      * and the http status code to the given value.
      * @param response the http response
      * @param httpStatusCode the http status code of the response
      * @param result the object to serialize to JSON
+     * @param type type token needed for gson
+     * @param <T> type of the result
      * @return a JSON string
      */
-    public static String json(Response response, int httpStatusCode, Object result) {
+    public static <T> String result(Response response, int httpStatusCode, ResultJson<T> result, TypeToken<T> type) {
+        response.type("application/json");
+        response.status(httpStatusCode);
+        ParameterizedType parameterizedType = newParameterizedTypeWithOwner(null, ResultJson.class, type.getType());
```

Comment:
Same concern as above, maybe we should create our own ParametrizedTypes like we do here:

https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/suse/scc/client/SCCClientUtils.java#L161-L180

It's verbose and kinda ugly, but I think it's safer this way.